jupyter-extension/src/components/DropdownCombobox/DropdownCombobox.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import {useCombobox} from 'downshift';
+
+export type DropdownComboboxProps = {
+  initialSelectedItem?: string | null;
+  items: string[];
+  onSelectedItemChange: (selectedItem: string | null) => void;
+  placeholder?: string;
+  testIdPrefix?: string;
+};
+
+export const DropdownCombobox: React.FC<DropdownComboboxProps> = ({
+  initialSelectedItem,
+  items,
+  onSelectedItemChange,
+  placeholder,
+  testIdPrefix,
+}) => {
+  const [inputItems, setInputItems] = React.useState(items);
+
+  testIdPrefix = testIdPrefix || '';
+  const {
+    isOpen,
+    getMenuProps,
+    getInputProps,
+    highlightedIndex,
+    getItemProps,
+    selectedItem,
+    selectItem,
+  } = useCombobox({
+    items: inputItems,
+    initialIsOpen: initialSelectedItem ? false : true,
+    initialSelectedItem: initialSelectedItem,
+    onInputValueChange: ({inputValue}) => {
+      setInputItems(
+        inputValue
+          ? items.filter((item) =>
+              item.toLowerCase().startsWith(inputValue.toLowerCase()),
```

Comment:
Is it possible to support matches in the middle of the string? It might be slower, but could be useful if there's many project/repo combos in the displayed list and a user starts searching with the repo name. Having to include the project name in every search might be cumbersome.

Reply:
Good idea, let me look into this!

Reply:
I pulled in the https://github.com/kentcdodds/match-sorter library to provide a much fuzzier search so I think that should work out here.

Reply:
Awesome, this works great!